chia/full_node/full_node.py

```diff
@@ -1363,15 +1364,13 @@ async def add_block_batch(
                 return (
                     False,
                     None,
-                    current_ssi,
-                    current_difficulty,
-                    prev_ses_block,
                     Err(pre_validation_results[i].error),
                 )
 
         agg_state_change_summary: Optional[StateChangeSummary] = None
         block_record = await self.blockchain.get_block_record_from_db(blocks_to_validate[0].prev_header_hash)
         for i, block in enumerate(blocks_to_validate):
+            assert cs.prev_ses_block is None or cs.prev_ses_block.height < block.height
```

Comment:
not that i mind but why do we need the assert?

Reply:
oh, while developing this patch I had a bug where this assert failed. If you don't make a copy to pass into the `validate_blocks_multiprocessing()`, then `prev_ses_block` may be updated prematurely

Reply:
do we have a way to force a copy to be passed ? 
maybe just copy it at the start of validate_blocks_multiprocessing and leave a comment

Reply:
I'm not sure we necessarily always want a copy. I think in all instances right now we probably do, but I believe adding batches of blocks will need to be split up a bit more, to be able to add blocks to the database concurrently with validating the next ones.
So, maybe once the dust settles on this we can fix things up a bit.